desktop/cmp/persistenceManager/impl/SaveDialogModel.ts

```diff
@@ -0,0 +1,86 @@
+import {FormModel} from '@xh/hoist/cmp/form';
+import {HoistModel, managed, PlainObject, TaskObserver, XH} from '@xh/hoist/core';
+import {lengthIs, required} from '@xh/hoist/data';
+import {makeObservable} from '@xh/hoist/mobx';
+import {PersistenceManagerModel} from '../PersistenceManagerModel';
+
+export class SaveDialogModel extends HoistModel {
+    parentModel: PersistenceManagerModel;
+
+    @managed saveTask = TaskObserver.trackLast();
+
+    @managed formModel = new FormModel({
+        fields: [
+            {
+                name: 'name',
+                rules: [
+                    required,
+                    lengthIs({max: 255}),
+                    ({value}) => {
+                        if (this.parentModel?.objects.find(it => it.name === value)) {
+                            return `An entry with name "${value}" already exists`;
+                        }
+                    }
+                ]
+            },
+            {name: 'description'}
+        ]
+    });
+
+    objStub: ObjStub;
+
+    get isAdd(): boolean {
+        return !!this.objStub?.isAdd;
+    }
+
+    constructor(parentModel: PersistenceManagerModel, objStub: ObjStub) {
+        super();
+        makeObservable(this);
+
+        this.parentModel = parentModel;
+        this.objStub = objStub;
+        this.formModel.init({
+            name: this.isAdd ? `` : `${objStub.name} (COPY)`,
+            description: this.isAdd ? `` : objStub.description
+        });
+    }
+
+    close() {
+        this.parentModel.closeSaveDialog();
+    }
+
+    async saveAsAsync() {
+        return this.doSaveAsAsync().linkTo(this.saveTask).catchDefault();
```

Comment:
Looks like this only catches exceptions thrown by `parentModel.refreshAsync()` -- not sure that's right